src/main/scala/viper/gobra/translator/Translator.scala

```diff
@@ -37,8 +39,36 @@ object Translator {
       case (t, transformer) => transformer.transform(t)
     }
 
-    config.reporter report GeneratedViperMessage(config.taskName, config.packageInfoInputMap(pkgInfo).map(_.name), () => transformedTask.program, () => transformedTask.backtrack)
+    config.reporter report GeneratedViperMessage(config.taskName, config.packageInfoInputMap(pkgInfo).map(_.name), () => sortAst(transformedTask.program), () => transformedTask.backtrack)
     transformedTask
   }
 
+  def sortAst(program: vpr.Program): vpr.Program = {
+    implicit def domainOrdering: Ordering[vpr.Domain] = Ordering.by(_.name)
+    implicit def domainFnOrdering: Ordering[vpr.DomainFunc] = Ordering.by(_.name)
+    implicit def domainAxOrdering: Ordering[vpr.DomainAxiom] = Ordering.by(FastPrettyPrinter.pretty(_))
+    implicit def fieldOrdering: Ordering[vpr.Field] = Ordering.by(_.name)
+    implicit def functionOrdering: Ordering[vpr.Function] = Ordering.by(_.name)
+    implicit def predicateOrdering: Ordering[vpr.Predicate] = Ordering.by(_.name)
+    implicit def methodOrdering: Ordering[vpr.Method] = Ordering.by(_.name)
+    implicit def extensionOrdering: Ordering[vpr.ExtensionMember] = Ordering.by(_.name)
```

Comment:
could we have only one `Ordering` for members and only introduce other orderings for other types which are not subtypes of members? In particular, I guess that `domainOrdering`, `functionOrdering`, `predicateOrdering`, `methodOrdering` and `extensionOrdering` could be merged into a single one.

Reply:
Great idea! Scala implicits seem to not work for subtypes but still better than how it is now